http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/http/service/RuntimeV2ServiceInterp.scala

```diff
@@ -520,99 +529,62 @@ class RuntimeV2ServiceInterp[F[_]: Parallel](config: RuntimeServiceConfig,
+
+      // Authorize: user has an active account and has accepted terms of service
+      _ <- authProvider.checkUserEnabled(userInfo)
+
+      // Authorize: get resource IDs the user can see
+
+      // TODO [] delegate authorization check for v2 resources to workspace-manager
+
+      // v2 runtimes are WSM-managed resources and their permissions are stored
+      // in relation to their WsmResourceSamResourceId
+      // HACK: accept any ID in the list of readable WSM resources as a valid
+      // readable runtime ID. Some of these IDs are for non-runtime resources.
+      readerRuntimeIds: List[String] <- authProvider
+        .getAuthorizedIds[RuntimeSamResourceId](isOwner = false, userInfo)
+        .flatMap(ids => F.pure(ids.map(_.resourceId).toList))
+      readerWsmIds: List[String] <- authProvider
+        .getAuthorizedIds[WsmResourceSamResourceId](isOwner = false, userInfo)
+        .flatMap(ids => F.pure(ids.map(_.resourceId).toList))
+      readerRuntimeOrWsmIds = readerRuntimeIds ++ readerWsmIds
+
+      readerWorkspaceIds: List[String] <- authProvider
+        .getAuthorizedIds[WorkspaceResourceSamResourceId](isOwner = false, userInfo)
+        .flatMap(ids => F.pure(ids.map(_.resourceId).toList))
+      ownerWorkspaceIds: List[String] <- authProvider
+        .getAuthorizedIds[WorkspaceResourceSamResourceId](isOwner = true, userInfo)
+        .flatMap(ids => F.pure(ids.map(_.resourceId).toList))
+      readerProjectIds: List[String] <- authProvider
+        .getAuthorizedIds[ProjectSamResourceId](isOwner = false, userInfo)
+        .flatMap(ids => F.pure(ids.map(_.resourceId).toList))
+      ownerProjectIds: List[String] <- authProvider
+        .getAuthorizedIds[ProjectSamResourceId](isOwner = true, userInfo)
+        .flatMap(ids => F.pure(ids.map(_.resourceId).toList))
+
+      // Parameters: parse search filters from request
       (labelMap, includeDeleted, _) <- F.fromEither(processListParameters(params))
       excludeStatuses = if (includeDeleted) List.empty else List(RuntimeStatus.Deleted)
       creatorOnly <- F.fromEither(processCreatorOnlyParameter(userInfo.userEmail, params, ctx.traceId))
+
       runtimes <- RuntimeServiceDbQueries
-        .listRuntimesForWorkspace(labelMap, excludeStatuses, creatorOnly, workspaceId, cloudProvider)
+        .listAuthorizedRuntimes(
+          labelMap, // arbitrary key-value labels to filter by
+          excludeStatuses, // whether to filter out Deleted runtimes
+          creatorOnly, // whether to filter out runtimes user did not create
+          workspaceId, // whether to find only runtimes in a single workspace
+          cloudProvider, // Google | Azure
+          // Authorization scopes
+          readerRuntimeOrWsmIds,
+          readerWorkspaceIds,
+          ownerWorkspaceIds,
+          readerProjectIds,
+          ownerProjectIds
+        )
         .map(_.toList)
         .transaction
-      filteredRuntimes <-
-        if (creatorOnly.isDefined) {
```

Comment:
where is the `creator` branch covered now?

Reply:
Thanks for noting this omission 🥶 - I added code to cover this case.